UnityProject/Assets/Scripts/FluidPipe/PipeData.cs

```diff
@@ -143,7 +145,12 @@ public virtual void OnDisable()
 					{
 						Outputs.Remove(Pipe);
 						//Removes itself if it is connected to a network and its a output for the network
-						Pipe.OnNet.RemoveEqualiseWith(this);
+
+						// this one probably require more work than just null check
+						if(Pipe.OnNet == null)
+							Logger.LogWarning("Pipe.OnNet == null", Category.Atmos);
+						else
+							Pipe.OnNet.RemoveEqualiseWith(this);
```

Comment:
NullReferenceException: Object reference not set to an instance of an object
Pipes.PipeData.OnDisable () (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/PipeData.cs:146)
Pipes.MonoPipe.OnDisable () (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/MonoPipe.cs:33)